app/src/main/java/com/example/whatsapp_clone/Views/MainActivity.java

```diff
@@ -141,10 +147,129 @@ public boolean onQueryTextChange(String newText) {
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
         if (item.getItemId() == R.id.settings) {
-            Toast.makeText(this, "in Settings", Toast.LENGTH_LONG).show();
+            showSettingsBottomSheet();
             return true;
         }
         return super.onOptionsItemSelected(item);
     }
 
+
+    private void showSettingsBottomSheet() {
+        View bottomSheetView = getLayoutInflater().inflate(R.layout.settings_bottom_sheet, null);
+        BottomSheetDialog bottomSheetDialog = new BottomSheetDialog(this);
+        bottomSheetDialog.setContentView(bottomSheetView);
+
+        // Calculate the desired height for the bottom sheet
+        int screenHeight = getResources().getDisplayMetrics().heightPixels;
+        int desiredHeight = (int) (screenHeight * 0.75);
```

Comment:
is this for the up to fit in different screen sizes? will it work for landscape mode? like when the screen is vertical?

Reply:
yes the height is determined based on the device orientation